test/test_filenames.py

```diff
@@ -0,0 +1,54 @@
+import os
+import pytest
+from pathlib import Path
+
+dir_list = [{
+    "type": "directory",
+    "name": "FYs",
+    "children": [
+        {
+        "type": "directory",
+        "name": "FY20",
+        "children": [
+            {
+            "type": "file",
+            "name": "letterofresignation_2020_10_04.docx"
+            },
+            {
+            "type": "file",
+            "name": "letterofresignation_2020_10_31.docx"
+            }
+        ]
+        },
+        {
+        "type": "file",
+        "name": "README.txt"
+        }
+    ]
+    }
+]
+
+def tree_builder(path, path_list):
+    for item in path_list:
+        if item["type"] == "file":
+            path.join(item["name"]).write('CONTENT')
```

Comment:
An idea for a future improvement would be to write (or give the option to write) random bytes rather than a fixed string. We could also provide 'content' or an 'empty' option in the json, which would let us create duplicate files when necessary